packages/data-context/src/sources/RelevantRunsDataSource.ts

```diff
@@ -145,10 +145,11 @@ export class RelevantRunsDataSource {
         continue
       }
 
-      if (run.status === 'RUNNING') {
-        allRuns.push(run.commitInfo.sha)
-      } else {
+      if (!allRuns.includes(run.commitInfo.sha)) {
```

Comment:
We should just make `allRuns` into a `Set`, then do `[...allRuns]` at the end - this would ensure no duplicates, and avoid this conditional entirely 😇 

I will add this in soon.